### DISCUSSION TITLE
Some questions about smart crossfading

### DISCUSSION THREAD

**Discussion by GeertSurkijn** [AzuraCast/AzuraCast#4211] [Q&A]
Hello, 
I am struggling with my smart crossfading settings. I have 2 questions, 1) is there a way to have a look at liquidsoap/src/operators/cross.ml ? I am not that familiar with docker, so was wondering where and how I can have a look (not change) at the content, I think it is not the default of liquidsoap? 2) Mainly for understanding this log output (the values of Analysis, which one is old, which one is new? I guess left is old?):

2021/05/25 10:15:57 [ffmpeg:3] [mp3float @ 0x7fd43c4f1dc0] Could not update timestamps for skipped samples.
2021/05/25 10:15:57 [cross_38980:3] Analysis: -19.724429dB / -17.828688dB (1.78s / 1.78s)
2021/05/25 10:15:57 [cross_38980:3] Not enough data for crossing.

I have most of the time the message: Not enough data for crossing, and I do not get it why. 
I also get a lot of these: 2021/05/25 10:24:56 [cross_38980:3] Analysis: -32.458075dB / -infdB (0.00s / 0.04s) Does this mean there is something wrong with analysing the volume on the right value? (-infdB)?

Thnx (Sam broadcaster user for years, trying to switch to Azuracast)

**Comment by Vaalyn** [comment]
1. You can see the source for the `cross` operator of Liquidsoap here on the Liquidsoap GitHub Repository: https://github.com/savonet/liquidsoap/blob/main/src/operators/cross.ml
2. Yep, left should be old, and right new track
